### PR TITLE
Document input/output resource attribute

### PR DESCRIPTION
**Pull request by soyuka**
https://github.com/api-platform/core/pull/2235 ping @silverbackdan

### CODE REVIEW

File: core/dto.md [hunk 1]
```diff
@@ -350,3 +350,73 @@ use Swagger\Annotations as SWG;
      */
     public function __invoke()
 ```
+
+### Specify an input and/or output class
```
Comment (dunglas) [inline core/dto.md]:
```suggestion
### Specify an Input and/or Output Class
```

File: core/dto.md [hunk 2]
```diff
@@ -350,3 +350,73 @@ use Swagger\Annotations as SWG;
...
+// api/src/Entity/Data.php
+
+namespace App\Entity;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use App\Dto\Input;
+use App\Dto\Output;
+
+/**
+ * @ApiResource(attributes={
```
Comment (dunglas) [inline core/dto.md]: You could use the shortcuts `inputClass` and `outputClass` directly

File: core/dto.md [hunk 3]
```diff
@@ -350,3 +350,73 @@ use Swagger\Annotations as SWG;
...
+class InputDataPersister implements DataPersisterInterface
+{
+    public function supports($data): bool
+    {
+        return $data instanceof Input;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
```
Comment (dunglas) [inline core/dto.md]: Could be remove in the example

File: core/dto.md [hunk 4]
```diff
@@ -350,3 +350,73 @@ use Swagger\Annotations as SWG;
...
+```php
+<?php
+
+namespace App\DataPersister;
+
+use ApiPlatform\Core\DataPersister\DataPersisterInterface;
+use App\Dto\Input;
+use App\Dto\Output;
+
+class InputDataPersister implements DataPersisterInterface
```
Comment (dunglas) [inline core/dto.md]:
```suggestion
final class InputDataPersister implements DataPersisterInterface
```

File: core/dto.md [hunk 5]
```diff
@@ -350,3 +350,73 @@ use Swagger\Annotations as SWG;
...
+
+        return $output;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove($data)
+    {
+        return null;
```
Comment (dunglas) [inline core/dto.md]:
```suggestion
        // TODO: implement removal
        return null;
```

**Comment by dunglas** [comment]
Shouldn't we remove the previous sections in the same file (https://api-platform.com/docs/core/dto/#how-to-use-a-dto-for-writing).
Using your solution is much better and cleaner, so we should recommend always using it.

**Comment by soyuka** [comment]
Yeah we should def remove the previous sections, but we need swagger support first :|. 
I'll add a deprecation sentence.

**Pull request merged**